dace/codegen/targets/fpga.py

```diff
@@ -1221,3 +1226,122 @@ def generate_host_function_boilerplate(self, sdfg, state, kernel_name,
         for arr_node in nested_global_transients:
             self._dispatcher.dispatch_allocate(sdfg, state, None, arr_node,
                                                None, host_code_stream)
+
+    def _generate_Tasklet(self, sdfg, dfg, state_id, node, function_stream,
+                          callsite_stream):
+
+        # TODO: this is mostly copy-pasta from the CPU-codegen. Refactor to
+        # inject hooks into the CPU codegen once the new codegen has been
+        # implemented.
```

Comment:
at this point, shouldn't you be able to override generate_tasklet_preamble/postamble and replace the keyword remover class with `DaceOpenCLKeywordRemover`? I'm not sure there is anything to override in the former (looks like an exact copy of the CPU version?); and the latter is possible because `unparse_tasklet` is now a method rather than a static function.

Reply:
`FPGACodegen` does not inherit from `CPUCodegen`, so I cannot override anything (that's why we have this `self._cpu_codegen` object floating around).

I believe we need new codegen for this, so we can have callbacks or change the inheritance structure. Do you agree?

Reply:
Did a more "correct" solution by changing `cpu.py` to accept a callback codegen.